src/sentry/grouping/enhancer/__init__.py

```diff
@@ -110,8 +111,8 @@ def parse_rust_enhancements(
             rust_enhancements = RustEnhancements.parse(input, RUST_CACHE)
 
         metrics.incr("rust_enhancements.parsing_performed", tags={"source": source})
-    except Exception:
```

Comment:
Are all errors that could be raised `RuntimeError`? In other words, should we be keeping both?

Reply:
We do raise them as such in the bindings. And from what I can see, the parsing methods only ever error for parse errors.
Apart from errors calling the bindings with wrong arguments, etc. I believe those are being raised as different error types, and they represent coding errors, so its better to raise a different kind of error there.